### DISCUSSION TITLE
Requête par Thierry #2: export json+csv

### DISCUSSION THREAD

**Discussion by esag-admin** [esag-swiss/iDig-Webapp#24] [Ideas]
En l'état actuel, dans la manipulation des données iDig, le plus laborieux est de faire des exports de données massives, càd les données de tous les secteurs/trenches d'un projet en un seul click, idéalement par type de données. But: import dans une autre bases de données, listings, etc. 
**Dans un 1er temps**: simple fonctionnalité d'export automatique csv+json de toutes les données par types incluant tous les trenches du projet. Il est ensuite aisé de filtrer les saisons/champs dans un fichier excel, par exemple.
**Dans un 2e temps**: fonctionnalité d'export des données telles que sélectionnées dans la web app (selon la sélection des trenches et des champs). À noter cependant que sélectionner manuellement tous les trenches et tous les champs dans la webapp sera également très laborieux... Peut-être ajouter un outil pour sélectionner/désélectionner tous les trenches et/ou tous les champs dans la webapp? 
**Dans un 3e temps**: avec l'export json+csv, il serait sans doute utile à terme de pouvoir exporter les données spatiales au format shape pour réutilisation dans un GIS. Mais c'est une fonctionnalité à développer dans le cadre plus général de la gestion des données spatiales dans la web app.

Pour info, voir un script développé par l'équipe anglaise de keros pour exporter les données de idig dans une database/GIS:
[https://github.com/archaeodata-code/Keros-Naxos-Seaways-iDigToDBase](https://github.com/archaeodata-code/Keros-Naxos-Seaways-iDigToDBase)

**Comment by AlZorba** [comment]
1 et 2 fait dans la PR https://github.com/esag-swiss/iDig-Webapp/pull/33
l'export se fait en json et fichier texte aves des séparateurs tab tout comme iDig app. Le CSV était moins évidant vu la nature des champs. Cela peut néanmoins être fait si nécessaire.
A voir aussi à l'usage si un export dump de tous les types à la fois est requis. 
Pour le 3 export des données en forma "SIG compatible" je crée une task dans la roadmap "export in GIS format" Je ne mentionne pas le format Shape, car il me semble que Geojson et plus indiqué. On regardera tout de même si le Shape peut être géré sans trop de difficultés.

**Comment by AlZorba** [comment]
3 fait aussi dans https://github.com/esag-swiss/iDig-Webapp/pull/68

**Comment by AlZorba** [comment]
3 fait aussi dans https://github.com/esag-swiss/iDig-Webapp/pull/68
